packages/@webex/plugin-cc/src/cc.ts

```diff
@@ -1,33 +1,73 @@
+/* eslint-disable no-console */
 import {WebexPlugin} from '@webex/webex-core';
 import {CCPluginConfig, IContactCenter, WebexSDK} from './types';
-import {WCC_API_GATEWAY} from './constants';
+import {
+  CC_EVENTS,
+  EVENT,
+  READY,
+  REGISTER_TIMEOUT,
+  SUBSCRIBE_API,
+  WCC_API_GATEWAY,
+} from './constants';
+import WebSocket from './WebSocket';
 
 export default class ContactCenter extends WebexPlugin implements IContactCenter {
   namespace = 'ContactCenter';
   $config: CCPluginConfig;
   $webex: WebexSDK;
   wccApiUrl: string;
-
+  webSocket: WebSocket;
+  ciUserId: string;
   constructor(...args) {
     super(...args);
     // @ts-ignore
     this.$config = this.config;
     // @ts-ignore
     this.$webex = this.webex;
+    this.$webex.once(READY, () => {
+      this.webSocket = new WebSocket({
+        parent: this.$webex,
+      });
+    });
   }
 
-  register(success: boolean): Promise<string> {
-    this.wccApiUrl = this.$webex.internal.services.get(WCC_API_GATEWAY); // Added this change for Ravi's PR, he will move this under different function if needed.
-    // TODO: Mercury Subsciption code should be added as part of this function
+  private establishConnection(): void {
+    const datachannelUrl = `${this.wccApiUrl}${SUBSCRIBE_API}`;
+
+    const connectionConfig = {
+      force: this.$config?.force ?? true,
+      isKeepAliveEnabled: this.$config?.isKeepAliveEnabled ?? false,
+      clientType: this.$config?.clientType ?? 'WxCCSDK',
+      allowMultiLogin: this.$config?.allowMultiLogin ?? true,
+    };
+
+    this.webSocket.subscribeAndConnect({
+      datachannelUrl,
+      body: connectionConfig,
+    });
+  }
+
+  register(): Promise<string> {
```

Comment:
Add jsdoc description here as this is going to be a public method

Reply:
Addressed